Project.toml

```diff
@@ -9,13 +9,15 @@ Markdown = "d6f4376e-aef5-505a-96c1-9c027394607a"
 Pkg = "44cfe95a-1eb2-52ea-b672-e2afdf69b78f"
 Printf = "de0858da-6303-5e67-8744-51eddeeeb8d7"
 REPL = "3fa0cd96-eef1-5676-8a61-b3b8758bbffb"
+TerminalRegressionTests = "98bfdc55-cc95-5876-a49a-74609291cbe0"
```

Comment:
It seems that TerminalRegressionTests, which [requires `VT100.PTY`](https://github.com/JuliaDebug/TerminalRegressionTests.jl/blob/ff1e668a0bf925049596662bfbd20f2cf8ddb265/src/TerminalRegressionTests.jl#L47) is [only available on unix system](https://github.com/Keno/VT100.jl/blob/894805f60f439347b1796c5b76eac741448f1d8c/src/VT100.jl#L660-L666). 

Any ideas how to proceed this so that it can works on windows? (Currently the precompilation stage would trigger TerminalRegressionTests error). Maybe this is more of a `TerminalRegressionTests` issue?

One way I can think of is to:

- in VT100.jl introduce an `AbstractPTY` type
- in TerminalRegressionTests.jl, use `AbstractPTY` instead of `PTY`, and move compilation error to runtime error. Also provide a `is_functional()` or-so function for users to query if this package is usable.
- in OhMyREPL, run precompilation only when `TerminalRegressionTests` is functional

cc: @Keno

Reply:
Oh, I guess I have to disable this on non Linux then for now..

Reply:
I might be able to do the same thing as we do to precompile the REPL stuff in Base.

Reply:
>  I assume that's why you mention the REPL stuff trick in Base?

Yes